File: Jack.Data/Repository/IGenericDao.vb
Public Interface IGenericDao(Of Tipo, ID)

    Function Insert(ByVal oTipo As Tipo) As Boolean
    Function Update(ByVal oTipo As Tipo) As Boolean
    Function Delete(ByVal oTipo As Tipo) As Boolean
    Function Find(ByVal Identifier As ID) As Tipo
    Function LoadAll() As IList(Of Tipo)

End Interface

File: Jack.Data/Repository/IGenericDao.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Data
{
    public interface IGenericDao<Tipo, ID>
    {

        bool Insert(Tipo oTipo);
        bool Update(Tipo oTipo);
        bool Delete(Tipo oTipo);
        Tipo Find(ID Identifier);
        IList<Tipo> LoadAll();

    }
}
